ratelimits/limiter.go

```diff
@@ -114,21 +127,28 @@ type Decision struct {
 // wait time before the client can make another request, and the time until the
 // bucket refills to its maximum capacity (resets). If no bucket exists for the
 // given limit Name and client id, a new one will be created WITHOUT the
-// request's cost deducted from its initial capacity.
+// request's cost deducted from its initial capacity. If the specified limit is
+// disabled, ErrLimitDisabled is returned.
 func (l *Limiter) Check(ctx context.Context, name Name, id string, cost int64) (*Decision, error) {
 	if cost < 0 {
 		return nil, ErrInvalidCostForCheck
 	}
 
 	limit, err := l.getLimit(name, id)
 	if err != nil {
+		if errors.Is(err, errLimitDisabled) {
+			return disabledLimitDecision, nil
+		}
 		return nil, err
 	}
 
 	if cost > limit.Burst {
 		return nil, ErrInvalidCostOverLimit
 	}
 
+	// Remove cancellation from the request context so that transactions are not
+	// interrupted by a client disconnect.
+	ctx = context.WithoutCancel(ctx)
```

Comment:
Should this line and the related comment move up to line 133 instead so that all check steps are performed outside the purview of context cancellation instead of just the network request over to redis? We're talking nanoseconds here so maybe this doesn't matter and is optimization for optimizations's sake.

Reply:
None of the logic above this point uses the context, so it doesn't actually matter -- there are no opportunities above here for a context cancellation to be noticed.